tdrs-backend/tdpservice/search_indexes/parsers/preparser.py

```diff
@@ -0,0 +1,169 @@
+"""Converts data files into a model that can be indexed by Elasticsearch."""
+
+import re
+import logging
+import argparse
+from cerberus import Validator
+from .util import get_record_type
+from . import tanf_parser
+# from .models import ParserLog
+from tdpservice.data_files.models import DataFile
+
+logger = logging.getLogger(__name__)
+logging.basicConfig(level=logging.DEBUG)
+
+
+def validate_header(datafile, data_type, given_section):
+    """Validate the header line of the datafile."""
+    """
+    https://www.acf.hhs.gov/sites/default/files/documents/ofa/transmission_file_header_trailer_record.pdf
+
+    DESCRIPTION		LENGTH	FROM	TO	COMMENT
+    Title		    6	1	6	Value	=	HEADER
+    YYYYQ	        5	7	11	Value	=	YYYYQ
+
+    Type 	        1	12	12	A=Active;	C=Closed;	G=Aggregate,	S=Stratum
+    State Fips	    2	13	14	"2	digit	state	code	000	a	tribe"
+    Tribe Code	    3	15	17	"3	digit	tribe	code	000	a	state"
+    Program	Type	3	18	20	Value	=	TAN	(TANF)	or	Value	=	SSP	(SSP-MOE)
+    Edit Indicator	1	21	21	1=Return	Fatal	&	Warning	Edits	2=Return	Fatal	Edits	only
+    Encryption      1	22	22	E=SSN	is	encrypted	Blank	=	SSN	is	not	encrypted
+    Update      	1	23	23	N	=	New	data	D	=	Delete	existing	data	U
+    QUARTERS:
+        Q=1	(Jan-Mar)
+        Q=2	(Apr-Jun)
+        Q=3	(Jul-Sep)
+        Q=4	(Oct-Dec)
+    Example:
+    HEADERYYYYQTFIPSSP1EN
+    """
+    logger.debug('Validating header row.')
+
+    section_map = {
+        'A': 'Active Cases',
+        'C': 'Closed Cases',
+        'G': 'Aggregate',
+        'S': 'Stratum',
+    }
+
+    # Validate the header row
+
+    # Apparently accessing files from db yields FieldFile which doesn't need to be opened 
+    # and I was running into errors passing it through redis
+    # So the below is commented out as it's obsolete for both tests and the actual parser
+    # with open(datafile, 'r') as f:
+    row = datafile.readline()
+
+    try:
+        header = {
+            'title':        row[0:6],
+            'year':         int(row[6:10]),
+            'quarter':      int(row[10:11]),
+            'type':         row[11:12],
+            'state_fips':   row[12:14],
+            'tribe_code':   row[14:17],
+            'program_type': row[17:20],
+            'edit':         row[20:21],
```

Comment:
I think `edit` and the state/tribe codes can probably be cast to int like `quarter` above.